### PR TITLE
implement grayscale_from_rgb

### PR DESCRIPTION
**Pull request by edgarriba**
#### Changes

- proves that we can just use raw tensor within keras_core
- implement `grayscale_from_rgb`
- add alias for temporal backward compatibility

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [x] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/color/gray.py [hunk 1]
```diff
@@ -32,52 +36,54 @@ def grayscale_to_rgb(image: Tensor) -> Tensor:
     return concatenate([image, image, image], -3)
 
 
-def rgb_to_grayscale(image: Tensor, rgb_weights: Tensor | None = None) -> Tensor:
+def grayscale_from_rgb(
```
Comment (ducha-aiki) [inline kornia/color/gray.py:39]: Shouldn't we remove the typing from image? np.ndarray is not a tensor

File: kornia/color/gray.py [hunk 2]
```diff
@@ -32,52 +36,54 @@ def grayscale_to_rgb(image: Tensor) -> Tensor:
...
+    # KORNIA_CHECK_NUM_CHANNELS(image, 3, axis=channels_axis)
 
     if rgb_weights is None:
-        # 8 bit images
-        if image.dtype == torch.uint8:
-            rgb_weights = torch.tensor([76, 150, 29], device=image.device, dtype=torch.uint8)
         # floating point images
-        elif image.dtype in (torch.float16, torch.float32, torch.float64):
-            rgb_weights = torch.tensor([0.299, 0.587, 0.114], device=image.device, dtype=image.dtype)
+        if "float32" in str(image.dtype):
```
Comment (ducha-aiki) [inline kornia/color/gray.py:72]: Just `float` maybe?

**Review by ducha-aiki — APPROVED** [review]

File: kornia/color/gray.py (hunk 1 above)
Comment (edgarriba) [reply kornia/color/gray.py:39]: i think we should map KerasTensor which is already an alias of the subset of supported tensor/ndarrays from the different frameworks. We can discover during the migration what's best and what give less headaches during development

File: tests/color/test_grayscale.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import numpy as np
+import pytest
+import tensorflow as tf
+import torch
+
+from kornia.color.gray import grayscale_from_rgb
+
+
+# TODO: make this a fixture
+def genrate_image_data(channel_first: bool = False):
```
Comment (johnnv1) [inline tests/color/test_grayscale.py:12]: I had imagined something like this to be added into `testing/`

Comment (johnnv1) [reply tests/color/test_grayscale.py:12]: i mean, this testing https://github.com/kornia/kornia/tree/keras-core/testing

which should live codes that are just utilities for tests, but isn't tests or fixtures itself

Comment (johnnv1) [reply tests/color/test_grayscale.py:12]: Like the BaseTester and assert close, it would not necessarily have to be part of the kornia package

**Pull request merged**